lib/checks/tables/th-has-data-cells.js

```diff
@@ -40,25 +40,21 @@ headers.forEach(header => {
 
 	const pos = tableUtils.getCellPosition(header, tableGrid);
 
-	// look for any column cell that has content and is not a column header
+	// ensure column header has at least 1 non-header cell
 	let hasCell = false;
 	if (tableUtils.isColumnHeader(header)) {
-		hasCell = tableUtils
-			.traverse('down', pos, tableGrid)
-			.some(
-				cell =>
-					axe.commons.dom.hasContent(cell) && !tableUtils.isColumnHeader(cell)
-			);
+		hasCell =
+			tableUtils
+				.traverse('down', pos, tableGrid)
+				.filter(cell => !tableUtils.isColumnHeader(cell)).length !== 0;
 	}
 
-	// look for any row cell that has content and is not a row header
+	// ensure row header has at least 1 non-header cell
 	if (!hasCell && tableUtils.isRowHeader(header)) {
-		hasCell = tableUtils
-			.traverse('right', pos, tableGrid)
-			.some(
-				cell =>
-					axe.commons.dom.hasContent(cell) && !tableUtils.isRowHeader(cell)
-			);
+		hasCell =
+			tableUtils
+				.traverse('right', pos, tableGrid)
+				.filter(cell => !tableUtils.isRowHeader(cell)).length !== 0;
```

Comment:
Wouldn't `.find()` be preferable here? `find` will halt after the first match where `filter` will always run through the full array.